### DISCUSSION TITLE
Minmal running example

### DISCUSSION THREAD

**Discussion by emoon** [dirkwhoffmann/vAmiga#484] [General]
Hi,

I'm trying to get vAmiga (own instance) up and running and what I currently have (running under linux) is something like this

```C++
    Amiga* amiga = new Amiga;
    amiga->mem.loadRomFromFile("/path/to/rom");
    amiga->mem.allocChip(512 * 1024);
    amiga->mem.allocSlow(512 * 1024);
    amiga->powerOn();
    amiga->run();
```

This starts running but quite soon I get this output:

```
Warning: The emulator is way too fast (0.204399).
Warning: The emulator is way too fast (0.201043).
Warning: The emulator is way too fast (0.204214).
Warning: The emulator is way too fast (0.205728).
Warning: The emulator is way too fast (0.203957).
vamiga/Utilities/RingBuffer.h:95: void util::RingBuffer<T, capacity>::write(T) [with T = Message; long int capacity = 64]: Assertion `!isFull()' failed.
```

I wonder what a minimal setup that works looks like.

**Comment by emoon** [comment]
I fixed the issue with RingBuffer running out of space (had to add a listener to the `messageQueue`) but I'm unsure about why the emulator is running too fast still.

**Comment by dirkwhoffmann** [comment]
> I'm unsure about why the emulator is running too fast still.

I'll check vAmiga BM in my Linux VM. Maybe I've screwed things up with my latest code changes...

**Comment by dirkwhoffmann** [comment]
Timing synchronisation is broken on Linux (vAmigaBM has the same issue). I'll look into the details...

**Comment by dirkwhoffmann** [comment]
Found it. Stupid mistake in the Linux implementation of sleepUntil() in Chrono.cpp:

```
void
Time::sleepUntil()
{
    (now() - *this).sleep();
}
```

must be

```
void
Time::sleepUntil()
{
    (*this - now()).sleep();
}
```

**Reply by emoon** [reply]
Great! I just got the first (super early version) of vAmiga running inside my debugger :)

![prodbg_vamiga](https://user-images.githubusercontent.com/350822/114172350-d6bda000-9935-11eb-9ad5-52b6104b0628.jpg)

**Reply by emoon** [reply]
I tested the fix above and it indeed fixes the issue. Thanks :)

**Reply by dirkwhoffmann** [reply]
Excellent! Great to see vAmiga being used in other projects.
